File: Python/백준/Silver/18110. solved．ac/solved．ac.py
import sys
input = sys.stdin.readline

def rounding(value):
    if value - int(value) >= 0.5:
        return int(value) + 1
    else:
        return int(value)

n = int(input())

if n == 0:
    print(0)
else:
    score = sorted([int(input()) for _ in range(n)])
    exception = rounding(n*0.15)
    exception_score = score[exception:n-exception]
    print(rounding(sum(exception_score) / len(exception_score)))

File: Java/백준/Silver/18110. solved．ac/solved．ac.java
import java.util.*;
import java.io.*;

public class Main {
    public static void main(String[] args) throws IOException {
        BufferedReader br = new BufferedReader(new InputStreamReader(System.in));

        int n = Integer.parseInt(br.readLine());

        if (n==0){
            System.out.println("0");
        }
        else {
            long[] scores = new long[n];

            for (int i = 0; i < n; i++) {
                scores[i] = Long.parseLong(br.readLine());
            }
            Arrays.sort(scores);

            int exception = (int) Math.round(n*0.15);
            long[] exception_score = new long[n-(2*exception)];
            for (int i = 0; i < n-(2*exception); i++) {
                exception_score[i] = scores[i+(2*exception)];
            }

            long sum = 0;

            for (int i = exception; i < n - exception; i++) {
                sum += scores[i];
            }

            double average = (double) sum / (n - 2 * exception);
            System.out.println(Math.round(average));
        }
    }
}